include/antler/system/utils.hpp

```diff
@@ -57,6 +58,39 @@ namespace antler::system {
       return s;
    }
 
+   inline static std::string exec_and_get_output(const std::string& cmd) {
+      std::array<char, 256> buffer;
+      std::string result;
+      std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
+      if (!pipe) {
+         system::error_log("internal failure, program {0} not found.", cmd);
+         return result;
+      }
+      while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+         result += buffer.data();
+      }
+      return result;
+   }
+
+   /// @return A tuple of ints indicating the version (major, minor, patch). Note that any pre-release or build metadata are ignored.
+   inline static auto get_cmake_ver() {
+
+      std::string ver = exec_and_get_output("cmake --version");
+
+      std::regex version_regex("cmake version (\\d+)\\.(\\d+)\\.(\\d+)");
+      std::smatch matches;
+      if (std::regex_search(ver, matches, version_regex)) {
+         int v1 = std::stoi(matches[1]);
+         int v2 = std::stoi(matches[2]);
+         int v3 = std::stoi(matches[3]);
+         return std::make_tuple(v1,v2,v3);
+      }
+      else {
+         system::info_log("CMake version {0} is incorrect", ver);
+         return std::make_tuple(0,0,0);
+      }

```

Comment:
How about if we extract this part of code to separate function and unit test it?

Currently i.e. if cmake will return "cmake version 0.0.4294967295" which exceeds `int` the result would be:
```
mikel@msi:~/repo/antler-proj/build/tools$ ./antler-proj build
[error_log] :: main.cpp:main:101 :: stoi
```
which is not terrible (because it won't happen), but it would be good to have such kind of unit tests.

Reply:
Yeah, that's a good idea.